qiskit/qobj/_validation.py

```diff
@@ -7,142 +7,17 @@
-    """Return a cascading explanation of the validation error in the form
-    of::
-
-        <validator> failed @ <subfield_path> because of:
-            <validator> failed @ <subfield_path> because of:
-                ...
-            <validator> failed @ <subfield_path> because of:
-                ...
-            ...
-
-    For example::
-
-        'oneOf' failed @ '<root>' because of:
-            'required' failed @ '<root>.config' because of:
-                'meas_level' is a required property
-
-    Meaning the validator 'oneOf' failed while validating the whole object
-    because of the validator 'required' failing while validating the property
-    'config' because its 'meas_level' field is missing.
-
-    The cascade repeats the format "<validator> failed @ <path> because of"
-    until there are no deeper causes. In this case, the string representation
-    of the error is shown.
-
-    Args:
-        err (jsonschema.ValidationError): the instance to explain.
-        level (int): starting level of indentation for the cascade of
-            explanations.
-
-    Return:
-        str: a formatted string with the explanation of the error.
-
-    """
-    lines = []
-
-    def _print(string, offset=0):
-        lines.append(_pad(string, offset=offset))
-
-    def _pad(string, offset=0):
-        padding = '  ' * (level + offset)
-        padded_lines = [padding + line for line in string.split('\n')]
-        return '\n'.join(padded_lines)
-
-    def _format_path(path):
-        def _format(item):
-            if isinstance(item, str):
-                return '.{}'.format(item)
-
-            return '[{}]'.format(item)
-
-        return ''.join(['<root>'] + list(map(_format, path)))
-
-    _print('\'{}\' failed @ \'{}\' because of:'.format(
-        err.validator, _format_path(err.absolute_path)))
-
-    if not err.context:
-        _print(str(err.message), offset=1)
-    else:
-        for suberr in err.context:
-            lines.append(_format_causes(suberr, level+1))
-
-    return '\n'.join(lines)
+from qiskit import QISKitError, validate_json_against_schema
 
 
 class QobjValidationError(QISKitError):
     """Represents an error during Qobj validation."""
     pass
 
 
-class _SummaryValidationError(QISKitError):
-    """Cut off the message of a jsonschema.ValidationError to avoid printing
-    noise in the standard output. The original validation error is in the
-    `validation_error` property.
-
-    Attributes:
-        validation_error (jsonschama.ValidationError): original validations
-            error.
-    """
-
-    def __init__(self, validation_error):
-        super().__init__(self._shorten_message(str(validation_error)))
-        self.validation_error = validation_error
-
-    def _shorten_message(self, message):
-        if len(message) > 1000:
-            return 'Original message too long to be useful: {}[...]'.format(message[:1000])
-
-        return message
+def validate_qobj_against_schema(qobj):
```

Comment:
Non-blocking: per our conversations, I think the idea is to manipulate QObject until right before sending it to the API, the point at where the validation should happen. We can leave this by now but could you open an issue asking for removing it and using an explicit conversion of the Qobject into a JSON dictionary, followed by an explicit validation against the proper schema?